include/ipfs/client.h

```diff
@@ -377,6 +377,48 @@ class Client {
       /** [in] Key name (local, user-friendly name for the key). */
       const std::string& key_name);
 
+  /** Publish an IPNS name attached to a given value.
+   *
+   * Implements
+   * https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/NAME.md#name.publish.
+   *
+   * An example usage:
+   * @snippet name.cc ipfs::Client::NamePublish
+   *
+   * @throw std::exception if any error occurs
+   *
+   * @since version 0.4.0 */
+  void NamePublish(
+      /** [in] Id (multihash) of the object to publish. */
+      const std::string& object_id,
+      /** [in] Name of the key to use. This is the local,
+       * human-friendly keyname */
+      const std::string& key_name,
+      /** [out] IPNS name Id (multihash) of the named object. */
+      std::string* name_id,
+      /** [in] Lifetime duration of the record. */
+      const std::string& lifetime = "24h",
+      /** [in] Duration of client's cache before new query. */
+      const std::string& ttl = "60s");
+
+  /** Resolve an IPNS name.
+   *
+   * Implements
+   * https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/NAME.md#name.resolve.
```

Comment:
Should be https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/NAME.md#nameresolve